Replicator/IncomingRev.cc

```diff
@@ -459,11 +460,11 @@ namespace litecore::repl {
         _blob = _pendingBlobs.end();
         _rev->trim();
 
-        // If IncomingRev is not the active Actor, this was called on the Puller's thread, so we can notify the
-        // Puller straight away.
-        // Otherwise, we will call `revWasHandled` later, in `Worker::afterEvent`.
-        if ( Actor::currentActor() != this ) _puller->revWasHandled(this);
-        else { _shouldNotifyPuller = true; }
+        // If insert was enqueued, the last code to fire will be in `afterEvent`, so delay return to
+        // Puller until `afterEvent`. Otherwise, notify Puller now so this IncomingRev can be recycled.
+        if ( _insertWasEnqueued.exchange(false) ) _shouldNotifyPuller = true;
```

Comment:
_shouldNotifyPuller is not atomically exchanged in afterEvent().

Reply:
Yes it is?